File: src/main/java/org/lineageos/sleepaudio/service/BootReceiver.java
package org.lineageos.sleepaudio.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            context.startForegroundService(new Intent(context, AudioService.class));
            Log.d("SleepAudio", "Boot complete, service requested.");
        }
    }
}

File: app/src/main/kotlin/org/lineageos/sleepaudio/service/BootReceiver.kt
package org.lineageos.sleepaudio.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import org.lineageos.sleepaudio.utils.Constants

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            Log.d(Constants.TAG, "Boot received, starting AudioService...")
            context.startForegroundService(Intent(context, AudioService::class.java))
        }
    }
}
